NMS-13076: BgpSessionDetector documenation (#3348)
OpenNMS/opennms: docs/modules/operation/pages/provisioning/detectors/BgpSessionDetector.adoc

= BGP Session Detector

> The BGP Service Detector can be used for detecting if a BGP-Session to a peering partner (peer-ip) is functional on devices with SNMP enabled. It will make a query against the RFC1269 MIB (`.1.3.6.1.2.1.15.3.1`) over SNMP to check a peering partner's status.

Reviewer: one line one sentence

> The BGP Service Detector can be used for detecting if a BGP-Session to a peering partner (peer-ip) is functional on devices with SNMP enabled. It will make a query against the RFC1269 MIB (`.1.3.6.1.2.1.15.3.1`) over SNMP to check a peering partner's status.

Reviewer 2: ```suggestion
Use the BGP Service Detector to detect if a BGP-Session to a peering partner (peer-ip) is functional on devices with SNMP enabled. It will query against the RFC1269 MIB (`.1.3.6.1.2.1.15.3.1`) over SNMP to check a peering partner's status.
```

> It checks that the return value is in the valid range of 1 (idle) to 6 (established).

Reviewer: Maybe an additional hint to make sure the detector figures out if a peering configuration exists ignoring the fact if it's really working or not, which is the job of the monitor.

== Detector facts

[options="autowidth"]
|===
| Implementation | `org.opennms.netmgt.provision.detector.snmp.BgpSessionDetector`
|===

== Configuration and Use

.Parameters for the BGP session detector

[options="header, autowidth"]
|===
| Parameter        | Description                                                                                        | Required | Default value

> | `bgpPeerIp`      | The the IPv4 peer ID to query for.                                          | required |

Reviewer 2: ```suggestion
| `bgpPeerIp`      | The IPv4 peer ID to query for.                                          | required |
```

Reviewer 2: No default value listed. Just making sure there is not one.

= BGP Session Detector

> Use the BGP Service Detector to detect if a BGP-Session to a peering partner (peer-ip) is configured.

Reviewer 2: > Use the BGP Service Detector to detect if a BGP Session to a peering partner (peer-ip) is configured.
